### PR TITLE
PDF reporter

### PR DESCRIPTION
**Pull request by scalvert**
This PR gets the basic structure for the PDF reporter working. It changes a few things:

1. The `--json` flag has been replaced with `--reporter`, with the options `stdout`, `json`, or `pdf`
2. The JSON output from a result is more constrained now, and contains both the task's `meta` and a `result` property.
3. The JSON results are aggregated now by `category`, then by `priority`. 

TODO:

- [x] Tests

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -5,14 +5,13 @@
     "ecmaVersion": 2017,
     "sourceType": "module"
   },
-  "plugins": ["@typescript-eslint", "filenames", "jest", "jsdoc", "prettier"],
+  "plugins": ["@typescript-eslint", "filenames", "jest", "prettier"],
```
Comment (scalvert) [inline .eslintrc]: Sorry, this was bugging me.

File: .eslintrc [hunk 2]
```diff
@@ -24,7 +23,8 @@
     "prettier/prettier": "error",
     "no-unused-vars": "off",
     "@typescript-eslint/no-unused-vars": "error",
-    "no-global-assign": ["error", { "exceptions": ["console"] }]
+    "no-global-assign": ["error", { "exceptions": ["console"] }],
+    "unicorn/prevent-abbreviations": "off"
```
Comment (scalvert) [inline .eslintrc]: So was this.

File: .eslintrc (hunk 1 above)
Comment (mahirshah) [reply .eslintrc]: thank you!

**Comment by mahirshah** [comment]
Overall looks good, although I was originally thinking that the reporters would actually each be a diff package, each being an oclif plugin that has a hook like registerReporter. The reporters would have a simply export a report method that takes in the task results. Thus the consumer of checkup would define in their `.checkuprc` something like `reporters: ['@checkup/pdf-reporter', '@foo/docx-reporter']`. This would let you have multiple reporters and keep your reporting consistent over multiple runs of the cli (which you can override via a cli option if necessary). Thoughts?

**Comment by scalvert** [comment]
That seems fine, but I think prematurely complicated. I'd rather get something working, and if we need to extract it we do it then. I almost think the parsers stuff is prematurely complicated right now...

**Pull request merged**